marimo/_save/stores/nix.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+import base64
+import dataclasses
+from hashlib import md5
+from typing import TYPE_CHECKING
+
+# import boto3
+import requests
+import zstandard as zstd
+
+from marimo._save.stores.store import Store
+
+if TYPE_CHECKING:
+    from marimo._save.cache import Cache
+    from marimo._save.hash import HashKey
+    from marimo._save.loaders import BasePersistenceLoader as Loader
+
+
+def compress_blob(blob: bytes, level: int = 3) -> bytes:
+    compressor = zstd.ZstdCompressor(level=level)
+    return compressor.compress(blob)
+
+
+def decompress_blob(blob: bytes) -> bytes:
+    decompressor = zstd.ZstdDecompressor()
+    return decompressor.decompress(blob)
+
+
+# TODO: Fix, abstract from just Cachix
+class NixRemoteStore(Store):
+    """
+    A remote Nix store implementation that interacts with Cachix.org.
+
+    This implementation mimics the API of FileStore but uses HTTP requests
+    to put and get serialized NAR files to/from a Cachix binary cache.
+    """
+
+    def __init__(
+        self,
+        cache_name: str = "marimo",
+        auth_token: str = "",
+        base_url: str = "https://app.cachix.org/api/v1",
```

Comment:
is this on the free 5gb for open-source?

we also have http://cache.marimo.app/ (i can make the API look similar)

Reply:
Yup, one of the free personal use tokens.

Started playing with this thinking a tiny bit further out for run verification (nix has some interesting artifact signature ideas, and dependency tracking). But I might break it out and shelve it for the time.
`StaticStore`/ `BucketStore` for WASM might be the next best "Store" to explore instead.

I'm not convinced `cache.marimo.app` should leverage nix caching methods- but it might be a nice plug and play.